### PR TITLE
Nd starting UI

### PR DESCRIPTION
**Pull request by SaraDawner2000**
add bootstrap and navbar

### CODE REVIEW

**Comment by CodeX-y** [comment]
Looks good! Good job with your partials.

**Review by CodeX-y — APPROVED** [review]

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,2 +1,3 @@
 class ApplicationController < ActionController::Base
+  before_action :authenticate_user!
```
Comment (JoseMDO) [inline app/controllers/application_controller.rb:2]: Nice!

File: app/views/layouts/application.html.erb [hunk 2]
```diff
@@ -6,11 +6,28 @@
     <%= csrf_meta_tags %>
     <%= csp_meta_tag %>
 
+    <%= render partial: "shared/cdn_assets" %>
```
Comment (JoseMDO) [inline app/views/layouts/application.html.erb:9]: Great job adding your CDN assets to a partial! This helps keep the layout file clean :)

File: app/views/layouts/application.html.erb [hunk 3]
```diff
@@ -6,11 +6,28 @@
...
+
+
     <%= stylesheet_link_tag "application", "data-turbo-track": "reload" %>
     <%= javascript_importmap_tags %>
   </head>
 
   <body>
+
+    <%= render partial: "shared/navbar" %>
```
Comment (JoseMDO) [inline app/views/layouts/application.html.erb:18]: same as above

File: app/views/layouts/application.html.erb [hunk 4]
```diff
@@ -6,11 +6,28 @@
...
+
+    <div class="container">
+      <% if notice.present? %>
+        <%=render partial: "shared/flash", locals: { message: notice, css_class: "success" }%>
```
Comment (JoseMDO) [inline app/views/layouts/application.html.erb:22]: same as above

File: app/views/shared/_navbar.html.erb [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+      <span class="navbar-toggler-icon"></span>
+    </button>
+    <div class="collapse navbar-collapse" id="navbarSupportedContent">
+      <ul class="navbar-nav mr-auto">
+        <li class="nav-item dropdown">
+          <a class="nav-link dropdown-toggle" href="#" role="button" data-bs-toggle="dropdown" aria-expanded="false">
+            Resources
+          </a>
+          <div class="dropdown-menu">
+            <%= link_to 'Photos', photos_path, class: "dropdown-item" %>
```
Comment (JoseMDO) [inline app/views/shared/_navbar.html.erb:14]: Nice job using the helper methods!

File: lib/tasks/dev.rake [hunk 6]
```diff
@@ -7,17 +7,24 @@ task sample_data: :environment do
...
-    user.email = "#{user.username}@example.com"
-    user.password = "password"
-    user.private = [true, false].sample
-    user.save
+  usernames = Array.new(12) { Faker::Internet.username }.uniq #is the unique method neccessary, or are there controls in place in the Faker gem that prevent the same value being generated in the same opertaion
+
+  usernames << "alice"
+  usernames << "bob"
+
+  usernames.each do |username|
```
Comment (JoseMDO) [inline lib/tasks/dev.rake:15]: great job on your sample data!

**Review by JoseMDO — APPROVED** [review]
Everything looks great!

**Pull request merged**